statsd/service_check.go

```
@@ -0,0 +1,57 @@
+	return &ServiceCheck{
+		Name:   name,
+		Status: status,
+	}
+}
+
+// Check verifies that a service check is valid.
+func (sc *ServiceCheck) Check() error {
+	if len(sc.Name) == 0 {
+		return fmt.Errorf("statsd.ServiceCheck name is required")
+	}
+	if byte(sc.Status) < 0 || byte(sc.Status) > 3 {
```

golangci-lint (1.62.2)

SA4003: no value of type byte is less than 0

(staticcheck)